linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_data_disk_full_action/rule.yml

```diff
@@ -56,3 +56,18 @@ ocil: |-
     switch to single-user mode, execute a script,
     or halt when the disk is out of space:
     <pre>disk_full_action single</pre>
+
+template:
+    name: auditd_lineinfile
+    vars:
+        missing_parameter_pass: 'false'
+        parameter: disk_full_action
+        rule_id: {{{ rule_id }}}
+        value: halt
+        value@sle15: syslog
+        value@rhcos4: syslog
+        value_regex@ol8: 'syslog|single|halt'
+        value_regex@rhel8: 'syslog|single|halt'
```

Comment:
The rule is parametrized by variable `var_auditd_disk_full_action` (see https://github.com/ComplianceAsCode/content/blob/da0aba874fc23ce2acdbadbc46ddf93d46d313f3/linux_os/guide/system/auditing/configure_auditd_data_retention/var_auditd_disk_full_action.var). The OVAL check you removed was using this variable. The template doesn't use the variable so the generated OVAL won't use it. At the same time Bash and Ansible will use it. That introduces an inconsistency. Users that will be creating their profiles might change this value and they will wonder why the OVAL check didn't change and isn't affected. So this isn't acceptable for us.

I think that you have 2 options:

1. Don't use the template here, keep a static OVAL file instead.

2. Modify the `auditd_lineinfile` template so that the template could accept a variable. You can get inspired for example by the `grub2_bootloader_argument` template https://github.com/ComplianceAsCode/content/tree/master/shared/templates/grub2_bootloader_argument and for example rule `grub2_slub_debug_argument`.

Reply:
I think it is better to use the template, I fixed it so it uses the existing variable if value_regex is not given